File: Baekjoon/1202.js
const reader = {
    input : require('fs').readFileSync('input.txt').toString().split("\r\n"),
    // input : require('fs').readFileSync('/dev/stdin').toString().split("\n"),
    index : 0,
    readLine : function(){
        return this.input[this.index++];
    },
    readList : function(delimiter){
        delimiter = delimiter == undefined ? " " : delimiter;
        return this.readLine().split(delimiter).map(e=>parseInt(e));
    }
}
let [N,K] = reader.readList();
const Items = [];
const Bags = [];

File: Baekjoon/1202.py
import sys,heapq;

N,K = map(int, sys.stdin.readline().split());
jewels = [];
bags = [];
for _ in range(N):
    jewels.append(list(map(int, sys.stdin.readline().split())));
for _ in range(K):
    heapq.heappush(bags, int(sys.stdin.readline()));
jewels.sort(key=lambda e: e[1]/e[0],reverse=True);
answer = 0;
for w,v in jewels:
    print(w,v)